### PR TITLE
migrate pipeline to kafka

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/368

### What changes does this PR make to Grapl? Why?

This PR migrates the data pipeline from an SQS/S3 based messaging tier to Kafka on Confluent Cloud.

### How were these changes tested?

Automated tests

### CODE REVIEW

**Review by colin-grapl — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: src/proto/graplinc/grapl/api/event_source/v1beta1/event_source.proto [hunk 1]
```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package graplinc.grapl.api.event_source.v1beta1;
+
+import "graplinc/common/v1beta1/types.proto";
+
+message CreateEventSourceRequest {
```
Comment (colin-grapl) [inline src/proto/graplinc/grapl/api/event_source/v1beta1/event_source.proto:7]: Gonna need `message` level comments to satsify the linters

File: src/proto/graplinc/grapl/api/event_source/v1beta1/event_source.proto [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+message GetEventSourceResponse {
+  EventSource event_source = 1;
+}
+
+service EventSourceService {
+  // write an event source to the database
+  // returns a response containing the event_source_id
+  rpc CreateEventSource(CreateEventSourceRequest) returns (CreateEventSourceResponse);
```
Comment (colin-grapl) [inline src/proto/graplinc/grapl/api/event_source/v1beta1/event_source.proto:87]: If the EventSource already exists in the db, what do we return? ALREADY_EXISTS ? Would be nice to document that I think

File: src/rust/event-source/src/config.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    pub db_config: EventSourceDbConfig,
+}
+
+impl EventSourceConfig {
+    /// An alias for clap::parse, so that consumers don't need to
+    /// declare a dependency on clap
+    pub fn from_env_vars() -> Self {
+        Self::parse()
+    }
+}
```
Comment (colin-grapl) [inline src/rust/event-source/src/config.rs:19]: fwiw another way to solve this is to have your crate `pub use thing_to_export`. If you do `pub use clap::Parser` the consumer won't have to have clap in their toml. But this is a fine solution too.

File: src/proto/graplinc/grapl/api/event_source/v1beta1/event_source.proto (hunk 1 above)
Comment (jgrillo-grapl) [reply src/proto/graplinc/grapl/api/event_source/v1beta1/event_source.proto:7]: The lints appear to pass: https://buildkite.com/grapl/grapl-verify/builds/3733#0181a619-8e63-481c-b178-f3adc1ea62ca

File: src/rust/generator-dispatcher/tests/test_utils/context.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            .expect("could not configure tracer");
+
+        // register a subscriber
+        let filter = EnvFilter::from_default_env();
+        tracing_subscriber::registry()
+            .with(filter)
+            .with(log_layer)
+            .with(tracing_opentelemetry::layer().with_tracer(tracer))
+            .init();
+
```
Comment (colin-grapl) [inline src/rust/generator-dispatcher/tests/test_utils/context.rs:56]: I think adding a single tracing subscriber should be done using the `test-log` crate with `features=["trace"]`

**Review by colin-grapl — APPROVED** [review]
A few minor suggestions

File: src/rust/event-source/migrations/20220615205523_create_event_source_table.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE IF NOT EXISTS event_sources (
+    event_source_id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    display_name varchar(128) NOT NULL,
+    description varchar(1024) NOT NULL,
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/rust/event-source/migrations/20220615205523_create_event_source_table.sql:5]: ticketed: https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/964

File: src/rust/generator-dispatcher/tests/test_utils/context.rs (hunk 4 above)
Comment (jgrillo-grapl) [reply src/rust/generator-dispatcher/tests/test_utils/context.rs:56]: If we do that here the log messages in this method won't be logged.. I think? Anyway, I'll defer to @wimax-grapl for this one as well

File: nomad/rust-integration-tests.nomad [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+variable "integration_tests_kafka_sasl_username" {
+  type        = string
+  description = "The Confluent Cloud API key to configure integration test consumers with."
+}
+
+variable "integration_tests_kafka_sasl_password" {
+  type        = string
+  description = "The Confluent Cloud API secret to configure integration test consumers with."
+}
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply nomad/rust-integration-tests.nomad]: done

**Pull request merged**